src/ensembl/src/content/app/entity-viewer/state/general/entityViewerGeneralActions.ts

```diff
@@ -139,33 +137,33 @@ export const changeActiveGenomeId: ActionCreator<ThunkAction<
   });
 };
 
-export const updateEntityViewerActiveEnsObjectIds = createAction(
-  'entity-viewer/update-active-ens-object-ids'
+export const updateActiveEntityForGenome = createAction(
+  'entity-viewer/update-active-entity-ids'
 )<{ [objectId: string]: string }>();
 
-export const updateEnsObject: ActionCreator<ThunkAction<
+export const updateEntityId: ActionCreator<ThunkAction<
   void,
   any,
   null,
   Action<string>
->> = (activeEnsObjectId: string) => {
+>> = (activeEntityId: string) => {
   return (dispatch, getState: () => RootState) => {
     const state = getState();
     const activeGenomeId = getEntityViewerActiveGenomeId(state);
     if (!activeGenomeId) {
       return;
     }
-    const currentActiveEnsObjectIds = getEntityViewerActiveEnsObjectIds(state);
-    const updatedActiveEnsObjectIds = {
-      ...currentActiveEnsObjectIds,
-      [activeGenomeId]: activeEnsObjectId
+    const currentActiveEntityIds = getEntityViewerActiveEntityIds(state);
+    const updatedActiveEntityIds = {
+      ...currentActiveEntityIds,
+      [activeGenomeId]: activeEntityId
     };
 
-    const currentEnsObject = getEntityViewerActiveEnsObject(state);
+    const currentEntity = getEntityViewerActiveEntity(state);
 
-    dispatch(updateEntityViewerActiveEnsObjectIds(updatedActiveEnsObjectIds));
-    if (!currentEnsObject) {
-      dispatch(fetchEnsObject(activeEnsObjectId));
+    dispatch(updateActiveEntityForGenome(updatedActiveEntityIds));
+    if (!currentEntity) {
+      dispatch(fetchEnsObject(activeEntityId));
```

Comment:
So we are still fetching the ensObject here. Do you remember why we are doing it?

Reply:
ensObject is unused here. I've removed these lives.